recipes/categorical_anomaly.py

```diff
@@ -141,15 +177,24 @@ def create_model(self,auto_tune=None, **model_args):
 
 
     def predict(self, df = None):
+        """
+        Runs fit_predict on a users data, with their given inputs, and returns a df with the score (-1 for anomalous, 1 for normal data)
+        Attempts to use scaled data, if users haven't scaled their data, will default to normal data.
+
+        LODA models cannot be run with less than 2 metrics, therefore, when a user selects LODA, and only 1 metric,
+        it will automatically select all of the numeric features of the dataset.
+        """
         if df is None:
             df = self.df
-
-            if self.ScalerClass:
-                scaler = self.ScalerClass()
-                scaler = scaler.fit(df[[self.metrics_column]])
-                df[self.metrics_column] = scaler.transform(df[[self.metrics_column]])
-        score_df = df.copy()
-        if self.model_type == 'LODA':
+        try:
+            score_df = self.scaled_df.copy()
+        except:
+            score_df = self.df.copy()
+        if self.model_type == 'LODA' and (type(self.metrics_column) == list):
+            score_df['score'] = self.model.fit_predict(score_df[self.metrics_column])
+            score_df['anomaly'] = score_df['score'] <= -1
+            self.score_df = score_df
```

Comment:
this assignment should happen after both conditionals (or the below `elif` becomes void)

Reply:
I couldn't get this to work? Tried moving the assignment but then it would fail when you were using say Loda and a list of metrics, or it wouldn't work with a single metric in a list. Just changed the logic in the if statements, it's working now? I tried every combination - LODA, one metric (str, and list), LODA multiple metrics, others one metric (string and list), and others multiple metrics, and it seems to be working

Reply:
weird, given all conditions conclude with the same assignment back to the object property

```python
self.score_df = score_df
```